weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CompareEvaluationsPage/sections/ExampleCompareSection/exampleCompareSectionUtil.ts

```diff
@@ -461,24 +479,52 @@ export function useExampleCompareData(
       );
 
       if (adjacentRowsToFetch.length > 0) {
-        // we load the data into the cache, but don't trigger a re-render
-        await loadRowDataIntoCache(
-          adjacentRowsToFetch,
-          cachedRowData,
-          cachedPartialTableRequest,
-          getTraceServerClient
+        // Check if any adjacent row data is available in comparison results
+        const adjacentRowsFromResults = adjacentRowsToFetch.filter(
+          digest => state.loadableComparisonResults.result?.inputs?.[digest]
+        );
+
+        // Add data from comparison results to cache
+        adjacentRowsFromResults.forEach(digest => {
+          const inputData =
+            state.loadableComparisonResults.result!.inputs[digest].val;
+          cachedRowData.current[digest] = inputData;
+        });
+
+        // Fetch remaining adjacent rows from table
+        const rowsToFetchFromTable = adjacentRowsToFetch.filter(
+          digest => !adjacentRowsFromResults.includes(digest)
         );
+
+        if (
+          rowsToFetchFromTable.length > 0 &&
+          cachedPartialTableRequest.current
```

Comment:
sort of ugly, but if current is null, then you hit a cache hit on the first one, and skipped loading... so you need to do

```
      if (!cachedPartialTableRequest.current) {
        cachedPartialTableRequest.current = await makePartialTableReq(
          state.summary.evaluations,
          filteredRows,
          targetIndex,
          getTraceServerClient
        );
      }
```

This function is going to be messy mvp